logstash-core/spec/logstash/instrument/wrapped_write_client_spec.rb

```diff
@@ -18,39 +18,54 @@
 
   subject { described_class.new(write_client, pipeline, metric, plugin) }
 
+  def threaded_read_client(read_client)
+    Thread.new(read_client) do |_read_client|
```

Comment:
since the client here doesn't need to be copied or modified, I see no use of passing it as variable into the thread.

Also, I see some inconsistent use of the variables passed this way into the threads, I will point them out below